asn1/LedgerLedgerLotocol.asn

```diff
@@ -0,0 +1,135 @@
+LedgerLedgerLotocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Address,
+    Timestamp
+    FROM InterledgerTypes
+;
+
+FreeRequest ::= SEQUENCE {
+    -- Used to associate requests and corresponding responses
+    -- MUST be either zero and non-repeating
+    -- If requestId = 0, the server MUST not send a response
+    requestId UInt128,
+    packet InterledgerPacket
+}
+
+FreeResponse ::= SEQUENCE {
+    -- Used to associate response to earlier FreeRequest
+    requestIdRef UInt128,
+    packet InterledgerPacket
+}
+
+PaidRequest ::= SEQUENCE {
```

Comment:
Why does there have to be a difference between prepare transfer and paid request? Couldn't you prepare a transfer and put a request in the side protocol data (or in the ILP packet, instead of an ILP payment packet)?